components/monitors/cu_consolemon/src/lib.rs

```diff
@@ -282,6 +288,147 @@ struct UI {
     sysinfo: String,
     task_stats: Arc<Mutex<TaskStats>>,
     nodes_scrollable_widget_state: NodesScrollableWidgetState,
+    error_redirect: BufferRedirect,
+    debug_output: Arc<DebugLog>,
+}
+
+struct DebugLog {
+    debug_log: Mutex<Vec<String>>,
+    max_col: Mutex<u16>,
+}
+
+impl DebugLog {
+    fn new() -> Self {
+        Self {
+            debug_log: Mutex::new(Vec::new()),
+            max_col: Mutex::new(100),
+        }
+    }
+
+    fn push_logs<A: AsRef<str>>(&self, logs: A) {
+        let mut log_lines = self.debug_log.lock().unwrap();
+        let max_col = *self.max_col.lock().unwrap();
+        log_lines.push(logs.as_ref().to_string());
+        if log_lines.len() > max_col as usize {
+            log_lines.remove(0);
+        }
+    }
```

Comment:
Should we use VecDeque instead of Vec for debug_log for performance consideration?